File: src/1621.py
import heapq


def find_regular_numbers(n):
    # Base case: if n is 0, return an empty list
    if n == 0:
        return []

    # Initialize the min-heap with the first regular number
    regular_numbers = [1]
    heap = [1]

    # Set to keep track of numbers already seen
    seen = set()
    seen.add(1)

    # Generate regular numbers until we reach n
    while len(regular_numbers) < n:
        # Get the smallest number from the heap
        smallest = heapq.heappop(heap)

        # Generate new numbers by multiplying with 2, 3, and 5
        for i in [2, 3, 5]:
            new_num = i * smallest
            if new_num not in seen:
                heapq.heappush(heap, new_num)
                seen.add(new_num)

        # Append the smallest number to the result
        regular_numbers.append(heap[0])

    return regular_numbers


# Example usage
result = find_regular_numbers(10)  # Finding the first 10 regular numbers
print(result)

File: src/1621.js
// https://chat.openai.com/share/36e6db49-921c-4be7-b682-311832e83da2

function findRegularNumbers(n) {
  if (n === 0) return [];

  let regularNumbers = [1];
  let heap = [1];
  let seen = new Set([1]);

  while (regularNumbers.length < n) {
    let smallest = heap.shift();

    [2, 3, 5].forEach((i) => {
      let newNum = i * smallest;
      if (!seen.has(newNum)) {
        heap.push(newNum);
        seen.add(newNum);
      }
    });

    heap.sort((a, b) => a - b); // Min-heapify
    regularNumbers.push(heap[0]);
  }

  return regularNumbers;
}

// Example usage
console.log(findRegularNumbers(10)); // Finding the first 10 regular numbers
